### PR TITLE
Datepicker improvements

### PR DESCRIPTION
**Pull request by x1unix**
* Allow hiding input inside the datepicker component using `headless` property
* Mark `isOpened` as property to allow to set datepicker status outside
* Allow embedding nested content inside the datepicker using `<ng-content>`

### CODE REVIEW

File: src/ng-datepicker/ng-datepicker.component.ts [hunk 1]
```diff
@@ -227,6 +239,11 @@ export class NgDatepickerComponent implements ControlValueAccessor, OnInit, OnCh
     }
 
     const input = this.elementRef.nativeElement.querySelector('.ngx-datepicker-input');
+
+    if (input == null) {
```
Comment (jkuri) [inline src/ng-datepicker/ng-datepicker.component.ts]: `if (!input) {` is better written I think.

Comment (x1unix) [reply src/ng-datepicker/ng-datepicker.component.ts]: I don’t think so because if input will be ‘true’ - it will pass this condition

Comment (jkuri) [reply src/ng-datepicker/ng-datepicker.component.ts]: okay, I missed that. sorry

**Comment by x1unix** [comment]
@jkuri, I don't know how to do that.

**Comment by jkuri** [comment]
yo Denis. the easiest way would be
```
git reset 9b05b1a3f228d3a86c4a006244894920077d05e4 # reset to last commit before yours
git add .
git commit -m "feat(input): allow to disable input"
git push origin master -f # you need to force push at this point since you were changing history
```
and that would be it. also, please bump version in `package.json` so I can publish this to npm afterwards. Thanks!

**Comment by x1unix** [comment]
@jkuri, done.

**Comment by jkuri** [comment]
thanks!

**Comment by x1unix** [comment]
@jkuri, also I plan to separate calendar from date-picker component to make this plugin more flexible. For example, you can embed the calendar anywhere, not just near the input and customise calendar positioning using your own stylesheets without workarounds.

**Comment by jkuri** [comment]
sounds good to me, thanks. just please update readme in case of changes in behavior or usage.

**Pull request merged**